### ISSUE TITLE
Running `npx eslint {path}` results in command failure.

### ISSUE THREAD

**Issue by jeffhou** [eslint/eslint#14079]
**Tell us about your environment**

* Node version: v14.15.3
* npm version: v7.5.2
* Local ESLint version: v7.19.0 (Currently used)
* Global ESLint version: Not found
* OS - macOS 11.1 (M1)

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  'env': {
    'browser': true,
    'commonjs': true,
    'es2021': true,
    'node': true,
  },
  'extends': [
    'plugin:vue/essential',
    'google',
  ],
  'parserOptions': {
    'ecmaVersion': 12,
  },
  'plugins': [
    'vue',
  ],
  'rules': {
  },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I installed ESLint using `npm install eslint`, set up ESLint using `npx eslint --init`, and then ran `npx eslint README.md`.

```bash
npx eslint README.md
```

**What did you expect to happen?**
I expected it the output to be
```bash
/Users/project_path/README.md
  1:1  error  Parsing error: Unexpected character '#'

✖ 1 problem (1 error, 0 warnings)
```

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
/Users/project_path/README.md
  1:1  error  Parsing error: Unexpected character '#'

✖ 1 problem (1 error, 0 warnings)

npm ERR! code 1
npm ERR! path /Users/project_path
npm ERR! command failed
npm ERR! command sh -c eslint "README.md"

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/me/.npm/_logs/2021-02-08T02_46_03_273Z-debug.log
```
It looks like some part of ESLint is also doing a `sh -c eslint "README.md"` inside the command. However, `eslint` isn't a command that can run by itself since I installed it using npm. It has to be run using `npx eslint`.

This issue doesn't occur when I run ESLint on a file without any errors/warnings.

**Are you willing to submit a pull request to fix this bug?**
I unfortunately don't know what is causing the issue or how to fix this bug.

**Comment by aladdin-add** [comment]
officially eslint only support js files. you will need to use https://github.com/eslint/eslint-plugin-markdown to lint md files.
